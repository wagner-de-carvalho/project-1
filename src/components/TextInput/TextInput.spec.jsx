import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import { TextInput } from '.';
import userEvent from "@testing-library/user-event";

describe('<TextInput />', () => {
    it('should call handleChange function on each key pressed', () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn} />);

        const input = screen.getByPlaceholderText(/type here your search/i)
        const inputValue = 'test'

        userEvent.type(input, inputValue);
        expect(input.value).toBe(inputValue)
        expect(fn).toHaveBeenCalledTimes(inputValue.length)

    })

    it('should have a value of searchValue', () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn} searchValue={'test'} />);
        const input = screen.getByPlaceholderText(/type here your search/i)

        expect(input).toBeInTheDocument()
        expect(input.value).toBe('test')
    })

    it('should match snapshot', () => {
        const fn = jest.fn();
        const { container } = render(<TextInput handleChange={fn} />)
        expect(container.firstChild).toMatchSnapshot();
    })
})