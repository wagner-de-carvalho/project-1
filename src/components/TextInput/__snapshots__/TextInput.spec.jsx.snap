// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Type here your search"
  type="search"
  value=""
/>
`;
